SubredditService/Sources/SubredditDataSource.swift

```diff
@@ -57,10 +60,20 @@ public class SubredditDataSource {
             switch result {
             case .success(let subreddits):
                 self?.subreddits.append(contentsOf: subreddits)
+                self?.pagingFinished = self?.setPagingState(count: subreddits.count) ?? false
                 self?.updated?()
             case .error:
                 return
             }
         }
     }
 }
+
+extension SubredditDataSource {
+    func setPagingState(count: Int) -> Bool {
```

Comment:
Would call `computePagingState` since the function itself does not set anything. Also should be private.